portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +27,46 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  private static final Gson gson = new Gson();
+
+  private ArrayList<String> comments = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Jag!");
+
+    response.setContentType("application/json");
+    String json = convertToJsonUsingGson(comments);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comment = getParameter(request, "text-input", "");
+
+    response.setContentType("text/html");
+    response.getWriter().println(comment);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(ArrayList dataList) {
```

Comment:
we can actually just get rid of this method and just do GSON.toJson() directly where ever convertToJsonUsingGson() is currently used. This method doesn't do much right now, and GSON.toJson() is descriptive enough about what it does

Reply:
done.